activesupport/lib/active_support/current_attributes.rb

```diff
@@ -188,12 +188,12 @@ def method_added(name)
         end
     end
 
-    class_attribute :defaults, instance_writer: false, default: {}
+    class_attribute :defaults, instance_writer: false, default: {}.freeze
```

Comment:
Thank you for following this up!

I mistook the [guidance provided by the `class_attribute` documentation](https://edgeapi.rubyonrails.org/classes/Class.html#method-i-class_attribute) as clone-on-inherit instead of mutation-free overrides:

> Declare a class-level attribute whose value is inheritable by subclasses. > Subclasses can change their own value and it will not impact parent class.